src/Middleware/SpaServices.Extensions/src/StaticFiles/SpaStaticFilesExtensions.cs

```diff
@@ -110,8 +111,8 @@ public static void UseSpaStaticFiles(this IApplicationBuilder applicationBuilder
                 }
             }
 
-            
-            app.UseStaticFiles(staticFileOptions);
+            // We should only serve files when there is no endpoint.
+            app.MapWhen(context => context.GetEndpoint() == null, a => app.UseStaticFiles(staticFileOptions));
```

Comment:
This is funky, but should it be added as a setting to staticFileOptions? That would be more user friendly/discoverable.

Reply:
Maybe.... this isn't incredibly common to do. All of our guidance for static files is to put it ahead of routing/mvc. This would only come up in a very specialized use case like this.